### PR TITLE
Cauchy-point method.

### PR DESCRIPTION
**Pull request by crisely09**
I still need to make the tests, mainly because I think it's easier if it was a function instead of a class, and I still need to have the SubProblem class to make it work.

### CODE REVIEW

File: flik/trustregion/cauchy.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        grad: callable
+            Gradient of the objective function.
+        hess: callable
+            Hessian of the objective function.
+        radius: float
+            The trust region radius.
+        """
+        super().__init__(point, func, grad, hess, radius)
+        self.gradient = self.grad(self.point)
+        self.hessian = self.hess(self.point)
```
Comment (lelemmen) [inline flik/trustregion/cauchy.py]: You are setting extra attributes here, so I think they should be specified in the docstring as well. Maybe a suggestion for naming your `gradient` and `hessian` to `current_gradient` and `current_hessian` to distinguish more clearly between callable and vector/matrix?

File: flik/trustregion/cauchy.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    def get_scaling(self):
+        r"""Define the scaling-factor of the step.
+
+        :math:`\tao = 1` if :math:`g^T B g \leq 0`
```
Comment (lelemmen) [inline flik/trustregion/cauchy.py]: I believe the Greek letter is actually called tau instead of tao.

Comment (crisely09) [reply flik/trustregion/cauchy.py]: ohh yes, you are right, thanks!

File: flik/trustregion/cauchy.py (hunk 1 above)
Comment (crisely09) [reply flik/trustregion/cauchy.py]: OK, true, I can make that change.

**Pull request merged**